CMakeLists.txt

```diff
@@ -1814,6 +1823,13 @@ if(HPX_WITH_DOCUMENTATION)
     PATTERN ".git" EXCLUDE)
 endif()
 
+
+if(HPX_WITH_VIM_YCM)
+  configure_file(${CMAKE_SOURCE_DIR}/tools/vim/.ycm_extra_conf.py
+               ${HPX_WITH_VIM_YCM}/.ycm_extra_conf.py @ONLY)
```

Comment:
This is not in line with every other cmake generated file we have. please put the .ycm_extra_conf.py file into cmake/templates and rename it to .ycm_extra_conf.py.in (this avoids accidental usage of a "to be configured" script).

In addition, I would prefer to have the destination of the ycm config to be placed in the build directory under tools. The reason is that the ycm configuration seems to be highly build configuration dependant (using the compile commands), once you maintain multiple builds, it might get a little messy and you accidently overwrite it.
With that being said, since we keep the build and source trees mostly seperate, it would be nice to have an automated way of telling vim where to look, I can't think of a proper way, but that shouldn't keep the PR from being merged.